app/src/main/java/com/mertceyhan/bitcoinmarket/widget/MarketWidget.kt

```diff
@@ -0,0 +1,32 @@
+package com.mertceyhan.bitcoinmarket.widget
+
+import androidx.compose.runtime.Composable
+import androidx.datastore.preferences.core.Preferences
+import androidx.glance.*
+import androidx.glance.appwidget.GlanceAppWidget
+import androidx.glance.state.GlanceStateDefinition
+import androidx.glance.state.PreferencesGlanceStateDefinition
+import com.mertceyhan.bitcoinmarket.core.ui.theme.BitcoinMarketTheme
+import com.mertceyhan.bitcoinmarket.utils.extensions.darkModeEnabled
+import com.mertceyhan.bitcoinmarket.widget.receiver.MarketWidgetReceiver
+import com.mertceyhan.bitcoinmarket.widget.ui.BitcoinWidget
+
+class MarketWidget : GlanceAppWidget() {
+
+    override val stateDefinition: GlanceStateDefinition<*> = PreferencesGlanceStateDefinition
+
+    @Composable
+    override fun Content() {
+
+        val context = LocalContext.current
+        val prefs = currentState<Preferences>()
+        val currentPrice = prefs[MarketWidgetReceiver.currentPrice]
+        val changeRate = prefs[MarketWidgetReceiver.changeRate]
+        val isChangeRatePositive = prefs[MarketWidgetReceiver.isChangeRatePositive]
+
+        BitcoinMarketTheme(darkTheme = context.darkModeEnabled()) {
+            BitcoinWidget(currentPrice, changeRate, isChangeRatePositive)
```

Comment:
Can we create a UI model for the widget and put all of the parameters inside it? Like `BitcoinWidgetUiState`. Also, I think BitcoinWidget shouldn't take null parameters, we shouldn't deal with nullable fields on the UI.

https://developer.android.com/jetpack/guide/ui-layer?continue=https%3A%2F%2Fdeveloper.android.com%2Fcourses%2Fpathways%2Fandroid-architecture%23article-https%3A%2F%2Fdeveloper.android.com%2Fjetpack%2Fguide%2Fui-layer